api4/cloud.go

```diff
@@ -232,33 +232,46 @@ func validateBusinessEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if emailToValidate.Email != "" {
-		errValidatingEmail := c.App.Cloud().ValidateBusinessEmail(user.Id, emailToValidate.Email)
-		if errValidatingEmail != nil {
-			c.Err = model.NewAppError("Api4.valiateBusinessEmail", "api.cloud.request_error", nil, errValidatingEmail.Error(), http.StatusInternalServerError)
-			return
-		}
-		ReturnStatusOK(w)
+	errValidatingEmail := c.App.Cloud().ValidateBusinessEmail(user.Id, emailToValidate.Email)
+	if errValidatingEmail != nil {
+		c.Err = model.NewAppError("Api4.valiateBusinessEmail", "api.cloud.request_error", nil, errValidatingEmail.Error(), http.StatusInternalServerError)
 		return
 	}
+	ReturnStatusOK(w)
+}
 
-	// If no email was sent as body param, then validate current userAdmin email
-	errValidatingAdminEmail := c.App.Cloud().ValidateBusinessEmail(user.Id, user.Email)
+func validateWorkspaceBusinessEmail(c *Context, w http.ResponseWriter, r *http.Request) {
+	if c.App.Channels().License() == nil || !*c.App.Channels().License().Features.Cloud {
+		c.Err = model.NewAppError("Api4.validateWorkspaceBusinessEmail", "api.cloud.license_error", nil, "", http.StatusForbidden)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleWriteBilling) {
+		c.SetPermissionError(model.PermissionSysconsoleWriteBilling)
+		return
+	}
+
+	user, userErr := c.App.GetUser(c.AppContext.Session().UserId)
+	if userErr != nil {
+		c.Err = model.NewAppError("Api4.validateWorkspaceBusinessEmail", "api.cloud.request_error", nil, "", http.StatusInternalServerError)
+		return
+	}
 
-	// if the current admin email is not a valid email
-	if errValidatingAdminEmail != nil {
+	// get the cloud customer email to validate if is a valid business email
+	cloudCustomer, err := c.App.Cloud().GetCloudCustomer(user.Id)
```

Comment:
You should check that err is not empty here because if there was an error, `cloudCustomer` will surely be `nil` and the program with panic at the next line (`cloudCustomer.Email`)
Except for rare case, it's better to check for `err` immediately

Reply:
you are right! nice catch @JulienTant . My last commit places the error validation to the correct place.